File: command/java/Peel.java
package command.java;

public class Peel implements Action {
    private String vegetable;

    public Peel(String vegetable) {
        this.vegetable = vegetable;
    }

    @Override
    public void Doit() {
        System.out.println("Peeling the " + vegetable);
    }
}

File: command/rust/command/src/peel.rs
// peel.rs
use crate::customer::{Action, CustomerInteraction};

pub struct Peel {
    pub interaction: CustomerInteraction,
}

impl Action for Peel {
    fn execute(&self) {
        println!(
            "The waiter is peeling an orange for {}.",
            self.interaction.customer.name
        );
    }
}
